classes/task/email_certificate_task.php

```diff
@@ -111,27 +165,35 @@ public function execute() {
                      WHERE ci.customcertid = :customcertid";
             $issuedusers = $DB->get_records_sql($sql, ['customcertid' => $customcert->id]);
 
-            // Now, get a list of users who can access the certificate but have not yet.
-            $enrolledusers = get_enrolled_users(\context_course::instance($customcert->courseid), 'mod/customcert:view');
-            foreach ($enrolledusers as $enroluser) {
-                // Check if the user has already been issued.
-                if (in_array($enroluser->id, array_keys((array) $issuedusers))) {
-                    continue;
-                }
+            // Now, get a list of users who can Manage the certificate.
+            $userswithmanage = get_users_by_capability($context, 'mod/customcert:manage', 'u.id');
 
-                // Now check if the certificate is not visible to the current user.
-                $cm = get_fast_modinfo($customcert->courseid, $enroluser->id)->instances['customcert'][$customcert->id];
-                if (!$cm->uservisible) {
-                    continue;
-                }
+            // Get the context of the Custom Certificate module.
+            $cm = get_coursemodule_from_instance('customcert', $customcert->id, $customcert->course);
+            $context = \context_module::instance($cm->id);
 
-                // Don't want to email those with the capability to manage the certificate.
-                if (has_capability('mod/customcert:manage', $context, $enroluser->id)) {
+            // Now, get a list of users who can view and issue the certificate but have not yet.
+            // Get users with the mod/customcert:receiveissue capability in the Custom Certificate module context.
+            $userswithissue = get_users_by_capability($context, 'mod/customcert:receiveissue',
+            'u.id, username, firstname, lastname, email, firstnamephonetic, lastnamephonetic, middlename, alternatename');
```

Comment:
This is prone to have errors in the future if these fields ever change and we are hardcoding them. Please use `\core_user\fields::for_identity($context)->get_sql('u', true, '', '', false);`. You might have to look into this more as I am not too familiar with this API - see https://docs.moodle.org/dev/User_fields.

Reply:
In fact - does this even need to be included? Why not just leave the default?

Reply:
Agree with you, I removed them